### PR TITLE
Fixes and tweaks for 2.3.0

### PR DESCRIPTION
**Pull request by nk3nny**
### When merged this pull request will:
Added option to disable enhanced fleeing function
Added Share information MP compatible
Added Share information will share to all friendly sides

Fixed taskArtillery script exits when out of ammo

Tweaked hideInside to be more defensive
Tweaked comms so contact shouts are not always run through formation leaders
Tweaked fleeing much less intrusive
Tweaked fleeing gestures changed. HiA, HiB, HiC was a poor option
Tweaked ShareInformationTweaked range
Tweaked civilian priorities in civilian.fsm

Removed unused variable in civilian.fsm

As per https://github.com/nk3nny/LambsDanger/issues/81

### CODE REVIEW

File: addons/danger/functions/fnc_hideInside.sqf [hunk 1]
```diff
@@ -67,7 +67,7 @@ if (!(_buildings isEqualTo []) && { RND(0.05) }) then {
     //[_unit, ["DOWN"], true] call FUNC(gesture);
 
     // find cover
-    private _cover = nearestTerrainObjects [ _unit, ["BUSH", "TREE", "SMALL TREE", "HIDE"], 13, false, true ];
+    private _cover = nearestTerrainObjects [ _unit getpos [30, getDir _unit + 180], ["BUSH", "TREE", "SMALL TREE", "HIDE"], 15, false, true ];
```
Comment (jokoho48) [inline addons/danger/functions/fnc_hideInside.sqf]:
```suggestion
    private _cover = nearestTerrainObjects [ _unit getPos [30, getDir _unit + 180], ["BUSH", "TREE", "SMALL TREE", "HIDE"], 15, false, true ];
```

File: addons/danger/functions/fnc_shareInformation.sqf [hunk 2]
```diff
@@ -35,8 +35,7 @@ _unit setVariable [QGVAR(currentTask), "Share Information"];
 
 // find units
 private _groups = allGroups select {
-    local _x
-    && {side _x isEqualTo side _unit}
+    && {(side _x) getFriend (side _unit) > 0.55}
```
Comment (jokoho48) [inline addons/danger/functions/fnc_shareInformation.sqf]: I would use https://community.bistudio.com/wiki/BIS_fnc_sideIsFriendly for this.

Comment (nk3nny) [reply addons/danger/functions/fnc_shareInformation.sqf]: aha! Thats the one. I'll fix it

File: addons/danger/functions/fnc_react.sqf [hunk 3]
```diff
@@ -16,7 +16,7 @@
  *
  * Public: No
 */
-params ["_unit", "_pos"];
+params ["_unit", "_pos", ["_enemy", ObjNull]];
```
Comment (jokoho48) [inline addons/danger/functions/fnc_react.sqf]:
```suggestion
params ["_unit", "_pos", ["_enemy", objNull]];
```

**Pull request merged**